### PR TITLE
Sms Workflow entity

### PR DESCRIPTION
**Pull request by aaronschachter**
Creates a `dosomething_sms_workflow` table to store SMS Reportback Workflows.

Next PR will query this table to actually create keywords and endpoints.

Refs https://jira.dosomething.org/browse/DOS-125

screen shot 2014-07-30 at 12 59 13 pm
screen shot 2014-07-30 at 12 59 42 pm

### CODE REVIEW

File: lib/modules/dosomething/dosomething_sms/dosomething_sms.admin.inc [hunk 1]
```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * @file
+ * Code for dosomething_sms admin functionality.
+ */
+
+function dosomething_sms_workflow_overview_page() {
+  $admin_path = DOSOMETHING_SMS_WORKFLOW_ADMIN_PATH;
+  $add_path = $admin_path . '/add';
+  $add_link = l(t('Add SMS Workflow'), $add_path);
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_sms/dosomething_sms.admin.inc]: `$add_link = l(t('Add SMS Workflow'), $admin_path . '/add');` ?

File: lib/modules/dosomething/dosomething_sms/dosomething_sms.admin.inc [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    t('Op'),
+  );
+
+  foreach (dosomething_sms_get_sms_workflows() as $workflow) {
+    $nid = $workflow->nid;
+    $rows[] = array(
+      $workflow->id,
+      $workflow->endpoint,
+      l($nid, 'node/' . $nid . '/reportbacks'),
+      $workflow->status ? t('Enabled') : t('Disabled'),
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_sms/dosomething_sms.admin.inc:26]: Does it work? I'd put it in parentheses, but that's ok if it works.

Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_sms/dosomething_sms.admin.inc:26]: It does, viewable in screenshot. I can update this though.

File: lib/modules/dosomething/dosomething_sms/dosomething_sms.admin.inc (hunk 1 above)
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_sms/dosomething_sms.admin.inc]: Good point - if nothing else critical I'll make this change in my next PR, as there's another one coming up soon.

File: lib/modules/dosomething/dosomething_sms/dosomething_sms.admin.inc [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      l(t('Edit'), $admin_path . '/' . $workflow->id . '/edit'),
+    );
+  }
+  return $output . theme('table', array(
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_sms/dosomething_sms.admin.inc]: I wouldn't overload return so much, but split it to `$output .= theme(…` , `return $output`;

Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_sms/dosomething_sms.admin.inc]: Good point.

File: lib/modules/dosomething/dosomething_sms/dosomething_sms.admin.inc (hunk 2 above)
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_sms/dosomething_sms.admin.inc:26]: No, leave it like this then.

**Comment by sergiitk** [comment]
:+1:

**Pull request merged**